stl/inc/random

```diff
@@ -329,7 +419,8 @@ _NODISCARD _Real _Nrand_impl(_Gen& _Gx) { // build a floating-point value from r
 
     if constexpr (_Has_static_min_max<_Gen>::value && _Minbits <= 64) {
         return _STD generate_canonical<_Real, _Minbits>(_Gx);
-    } else { // TRANSITION, for tr1 machinery only; Standard machinery can call generate_canonical directly
+    } else if constexpr (!is_integral_v<typename _Gen::result_type>) {

```

Comment:
Style: We conventionally prefer positive tests unless there's a clarity advantage otherwise (it doesn't even need to be a huge advantage, any small reason will do). Here, I see no reason - the order Standard, integral tr1, non-integral tr1 would be fine, and sorted in order of ascending wackiness, so let's avoid the negation.